### ISSUE TITLE
Collection Editor Add button is grayed out for a collection with no setter

### ISSUE THREAD

**Issue by masonwheeler** [xceedsoftware/wpftoolkit#1686]
```
class Foo
{
   public ObservableCollection<Bar> Stuff {get;} = new();
}
```

This is a standard, idiomatic way to set up collections in C#.  But when attempting to edit one in the Collection Editor, it considers *the collection's item list* to be read-only because *the property itself* is read-only.

Obviously this is not correct.  As long as the class implements `ICollection<T>`, the item list is editable.  If it implements `IList<T>`, it's indexable (and thus reorderable) as well.

**Comment by XceedBoucherS** [comment]
Hi,
Simply set the NewItemTypes property. This will tell which types can be added in the collection:
<xctk:CollectionControlButton x:Name="_collectionControlButton"
                                  ItemsSource="{Binding}" />

public partial class MainWindow : Window
  {
    public MainWindow()
    {
      this.DataContext = this;
      InitializeComponent();

      _collectionControlButton.NewItemTypes = new List<Type>() { typeof(Bar) };
    }

    public ObservableCollection<Bar> Stuff { get; } = new ObservableCollection<Bar>();
  }

  public class Bar
  {
    public string MyString
    {
      get;
      set;
    }
  }

**Comment by masonwheeler** [comment]
Yes, that can be done.  But it doesn't really address the issue.  If the exact same collection property is marked get/set, no `NewItemTypes` is needed.  The problem, then, isn't that the code isn't already capable of determining the element type of a collection; the problem is that it's confusing a read-only collection *property* for a read-only collection *item list.*

**Comment by XceedBoucherS** [comment]
Hi,

Well, if I take the sample I've paste earlier, set the property "Stuff" to be a get/set and remove the line:
_collectionControlButton.NewItemTypes = new List<Type>() { typeof(Bar) };
The CollectionEditor won't make the "Add" option available. The NewItemTypes must be set for the "Add" option to be available. You need to specify what type of object can be added. 
If for example you have the base class "Person" and the derive classes "Man" and "Woman"(which derive from "Person"), and your property is a collection of Person, you may want to Add Man and Woman, or only Man or only Woman. You need to specify it to the CollectionEditor. This is done with the NewItemTypes property.

**Comment by masonwheeler** [comment]
Thanks!

Could you get someone to look at [#1687](https://github.com/xceedsoftware/wpftoolkit/issues/1687), BTW?  It's been sitting there for almost 3 weeks with no attention...
